graveler/committed/tree/tree.go

```diff
@@ -0,0 +1,69 @@
+package tree
+
+import (
+	"github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/graveler/committed/sstable"
+)
+
+// part is the basic building stone of a tree
+// nolint: structcheck, unused
+type Part struct {
+	Name   sstable.ID
+	MaxKey graveler.Key
+}
+
+// Tree is a sorted slice of parts with no overlapping between the parts
+// nolint: structcheck, unused
+type Tree struct {
+	ID    graveler.TreeID
+	Parts []Part
+}
+
+// Repo is an abstraction for a repository of trees that exposes operations on them
+type Repo interface {
+	GetTree(treeID graveler.TreeID) (Tree, error)
```

Comment:
If tree is not an interface, better return a pointer. And receive it as a pointer in all other methods

Reply:
same for ValueRecord